time_manager/time_manager.F90

```diff
@@ -1147,73 +865,65 @@ subroutine time_assignment(time1, time2)
    time1%days    = time2%days
    time1%ticks   = time2%ticks
 end subroutine time_assignment
-! </SUBROUTINE>
-
-!-------------------------------------------------------------------------
-! <FUNCTION NAME="time_type_to_real">
-!   <OVERVIEW>
-!       Converts time to seconds and returns it as a real number
-!   </OVERVIEW>
-!   <DESCRIPTION>
-!       Converts time to seconds and returns it as a real number
-!   </DESCRIPTION>
-!   <TEMPLATE>
-!     time_type_to_real(time)
-!   </TEMPLATE>
-!   <IN NAME="time" UNITS="" TYPE="time_type" DIM="">
-!      A time interval.
-!   </IN>
 
+!> Converts time to seconds and returns it as a real number
 function time_type_to_real(time)
 
 real(kind=r8_kind)           :: time_type_to_real
 type(time_type), intent(in) :: time
 
 if(.not.module_is_initialized) call time_manager_init
 
-time_type_to_real = real(dble(time%days) * 86400.d0 + dble(time%seconds) + &
-     dble(time%ticks)/dble(ticks_per_second), kind=r8_kind)
+time_type_to_real = real(time%days, r8_kind) * 86400.e0_r8_kind + real(time%seconds, r8_kind) + &
+                    real(time%ticks, r8_kind)/real(ticks_per_second, r8_kind)
 
 end function time_type_to_real
 
 !> @brief Convert a real number of seconds into a time_type variable.
 !! @return A filled time type variable, and an error message if an
 !!         error occurs.
-function real_to_time_type(x,err_msg) result(t)
-  real,intent(in) :: x !< Number of seconds.
+function real8_to_time_type(x,err_msg) result(t)
+  real(r8_kind),intent(in) :: x !< Number of seconds.
   character(len=*),intent(out),optional :: err_msg !< Error message.
   type(time_type) :: t
   integer :: days
   integer :: seconds
   integer :: ticks
   character(len=128) :: err_msg_local
-  real,parameter :: spd = real(86400)
-  real :: tps
-  real :: a
-  tps = real(ticks_per_second)
+  real(r8_kind),parameter :: spd = 86400.0_r8_kind
+  real(r8_kind) :: tps
+  real(r8_kind) :: a
+  tps = real(ticks_per_second, r8_kind)
   a = x/spd
   days = safe_rtoi(a,do_floor)
-  a = x - real(days)*spd
+  a = x - real(days, r8_kind)*spd
   seconds = safe_rtoi(a,do_floor)
-  a = (a - real(seconds))*tps
+  a = (a - real(seconds, r8_kind))*tps
   ticks = safe_rtoi(a,do_nearest)
   if (.not. set_time_private(seconds,days,ticks,t,err_msg_local)) then
-    if (error_handler('function real_to_time_type',err_msg_local,err_msg)) then
+    if (error_handler('function real8_to_time_type',err_msg_local,err_msg)) then
       return
     endif
   endif
-end function real_to_time_type
+end function real8_to_time_type
+
+function real4_to_time_type(x, err_msg) result(t)
```

Comment:
would this function be needed if all reals are r8_kind in time_manager?

Reply:
Yeah it's for external routines, if a module is compiled with r4 it'll call this to get a new time type. It gets converted/rounded anyway by `safe_rtoi`